src/duk_debugger.c

```diff
@@ -37,8 +37,23 @@ DUK_LOCAL void duk__debug_do_detach1(duk_heap *heap) {
 	 * inside the detached callback.
 	 */
 
+	if (heap->dbg_detaching) {
+		return;
+	}
+
 	DUK_D(DUK_DPRINT("debugger transport detaching, marking transport broken"));
 
+	heap->dbg_detaching = 1;
+
+	/* avoid sending multiple Detach notifys */
+	if (heap->dbg_write_cb != NULL) {
+		duk_hthread *thr = heap->heap_thread;
+
```

Comment:
It might be zero during heap destruction - of course we shouldn't be attached at that point but since we do detach in heap destruction too, we're basically assuming that the destruction sequence is well ordered in this respect. The assert would communicate that intent :)